tests/test_main.py

```diff
@@ -44,3 +45,21 @@ def test_main_scan(self, mock_cfg, mock_env, mock_scanner):
         environment_instance.setup.assert_called_once()
         environment_instance.scanner.assert_called_once()
         mock_scanner.scan.assert_called_once()
+
+    @patch("py_sonar_scanner.scanner.Scanner")
+    @patch("py_sonar_scanner.__main__.Environment")
+    @patch("py_sonar_scanner.__main__.Configuration")
+    def test_main_scan_fail(self, mock_cfg, mock_env, mock_scanner):
```

Comment:
Given that logging depends on `debug` mode, I think it would be nice to have 2 versions of this test, with and without debug mode enabled, so that it's clear what is logged exactly in each case (I assume in one case only the exception name appears, and in the other, the stack trace appears? Then maybe we need some mock information for that as well).